pkg/types/interfacetests/codec_interface_tests.go

```diff
@@ -0,0 +1,280 @@
+package interfacetests
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/smartcontractkit/libocr/commontypes"
+	ocrTypes "github.com/smartcontractkit/libocr/offchainreporting2/types"
+	"math/big"
+
+	"github.com/smartcontractkit/chainlink-relay/pkg/types"
+)
+
+type EncodeRequest struct {
+	TestStructs  []TestStruct
+	ExtraField   bool
+	MissingField bool
+	TestOn       string
+}
+
+type ChainReaderInterfaceTester interface {
+	Setup()
+	Teardown()
+	Name() string
+	EncodeFields(request *EncodeRequest) ocrTypes.Report
+	CreateAccount(i int) []byte
+	GetChainReader() types.ChainReader
+
+	// IncludeArrayEncodingSizeEnforcement is here in case there's no way to have fixed arrays in the encoded values
+	IncludeArrayEncodingSizeEnforcement() bool
+
+	// SetLatestValue is expected to return the same bound contract and method in the same test
+	// Any setup required for this should be done in Setup.
+	// The contract should take a LatestParams as the params and return the nth TestStruct set
+	SetLatestValue(testStruct *TestStruct) (types.BoundContract, string)
+}
+
+const (
+	TestItemType       = "TestItem"
+	TestItemSliceType  = "TestItemSliceType"
+	TestItemArray1Type = "TestItemArray1Type"
+	TestItemArray2Type = "TestItemArray2Type"
+)
+
+// CreateChainReaderInterfaceTests uses TestStruct and TestStructWithSpecialFields
+func CreateChainReaderInterfaceTests(tester ChainReaderInterfaceTester) bool {
+	ctx := context.Background()
+	BeforeEach(tester.Setup)
+	AfterEach(tester.Teardown)
+	return Describe("Codec tests for "+tester.Name(), func() {
+		It("Encodes and decodes a single item", func() {
```

Comment:
I have personally always avoided this framework because this style of test naming doesn't make any sense to me in go. Is there a clean way to invoke individual sub-tests? Without e.g. replacing all the spaces with underscores?

Reply:
I'll try to remove the library, per your other comment :).  It is convention with these libraries to do this kind of naming.  I'll do subtests instead or something else.